### PR TITLE
Implement resolving regex capture groups

### PR DESCRIPTION
**Pull request by phanimarupaka**
@frankfarzan 

This PR is to implement the regex capture groups to put-value and put-comment for Search And Replace functionality. 

Flags renamed:
put-literal -> put-value
put-pattern -> put-comment

Users can leverage complex regex to match the fields and replace them with resolved capture groups.

### CODE REVIEW

File: internal/cmdsearch/searchreplace_test.go [hunk 1]
```diff
@@ -59,7 +59,7 @@ metadata:
 	},
 	{
 		name: "search replace by value",
```
Comment (frankfarzan) [inline internal/cmdsearch/searchreplace_test.go:61]: should this file be called putvalue_test.go for consistency?

File: internal/cmdsearch/searchreplace_test.go [hunk 2]
```diff
@@ -624,7 +624,7 @@ spec:
 	},
 	{
 		name: "put literal error",
-		args: []string{"--put-literal", "something"},
+		args: []string{"--put-value", "something"},
```
Comment (frankfarzan) [inline internal/cmdsearch/searchreplace_test.go:627]: is there a test for put-value containing capture groups?

File: internal/util/search/search.go [hunk 3]
```diff
@@ -204,83 +202,47 @@ func (sr *SearchReplace) putLiteral(object *yaml.RNode) error {
...
-
-	pattern := fieldValue
-	// derive the pattern from the field value by replacing setter values
-	// with setter name markers
-	// e.g. if field value is "my-project-foo", input PutPattern is "${project]-*", and
-	// value for setter "project" is "my-project",
-	// the pattern to be added as comment is ${project]-foo
-	for sn, sv := range settersValues {
-		pattern = strings.ReplaceAll(clean(pattern), clean(sv), clean(sn))
+	r := regexp.MustCompile(valueRegex)
```
Comment (frankfarzan) [inline internal/util/search/search.go:231]: does this panic if it doesn't compile?

File: internal/util/search/search_test.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	patternRegex string
+	expected     string
+	errMsg       string
+}
+
+var resolvePatternCases = []patternTest{
+	{
+		name:         "resolve pattern 1",
+		fieldValue:   "foo1-dev-bar1-us-east-1-baz1",
+		valueRegex:   "([a-z]+[0-9]+)-dev-([a-z]+[0-9]+)-us-east-1-([a-z]+[0-9]+)",
```
Comment (frankfarzan) [inline internal/util/search/search_test.go:22]: nit: you can use metacharacter `\w` for brevity

File: internal/cmdsearch/cmdsearch.go [hunk 5]
```diff
@@ -46,10 +46,10 @@ func NewSearchRunner(name string) *SearchRunner {
 			"value of the field by default without requiring start (^) and end ($) characters.")
 	c.Flags().StringVar(&r.ByPath, "by-path", "",
 		"Match by path expression of a field.")
-	c.Flags().StringVar(&r.PutLiteral, "put-literal", "",
+	c.Flags().StringVar(&r.PutValue, "put-value", "",
```
[the remark this answers is not in the document]
Comment (phanimarupaka) [reply internal/cmdsearch/cmdsearch.go:49]: Done.

File: internal/cmdsearch/searchreplace_test.go (hunk 2 above)
Comment (phanimarupaka) [reply internal/cmdsearch/searchreplace_test.go:627]: Added. Had a test with both put-value and put-comment but added another test with just put-value containing capture groups.

File: internal/util/search/search.go (hunk 3 above)
Comment (phanimarupaka) [reply internal/util/search/search.go]: Yes. Updated PR to properly wrap and throw error and added test.

File: internal/util/search/search_test.go (hunk 4 above)
Comment (phanimarupaka) [reply internal/util/search/search_test.go]: Done.

File: internal/cmdsearch/searchreplace_test.go (hunk 1 above)
Comment (phanimarupaka) [reply internal/cmdsearch/searchreplace_test.go:61]: this file contains tests for both search and also put-value, so felt it is better name. Renamed other file to putcomment_test.go as it contains only --put-comment cases.

**Review by frankfarzan — APPROVED** [review]

**Pull request merged**